spec/account_concern_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+RSpec.describe "Sessions", type: :request do
+  describe "GET /show" do
+    let(:response_body) { JSON.parse(response.body) }
+
+    context "when the user is not logged in" do
+      let(:headers) { {} }
+
+      it "shows the logged_in status and the 'HTTP_GOVUK_ACCOUNT_SESSION' header value" do
+        get show_session_path, headers: headers
+
+        expect(response).to have_http_status(:ok)
+        expect(response_body).to eq("logged_in" => false, "account_session_header" => nil)
+      end
+    end
+
+    context "when the user is logged in" do
+      let(:account_session_header) { "foo" }
+      let(:headers) { { "HTTP_GOVUK_ACCOUNT_SESSION" => account_session_header } }
```

Comment:
The request header is `GOVUK-Account-Session`, not `HTTP_GOVUK_ACCOUNT_SESSION` - the header renaming happens somewhere in the Rails middleware.

Reply:
Also, you could reduce a little bit of code duplication by restructuring these tests as

```ruby
let(:account_session_header) { nil }
let(:headers) { { "GOVUK-Account-Session" => account_session_header }.compact }

context "when the user is not logged in" do
  # ...
end

context "when the user is logged in" do
  let(:account_session_header) { "foo" }
  # ...
end
```

Reply:
> The request header is `GOVUK-Account-Session`, not `HTTP_GOVUK_ACCOUNT_SESSION` - the header renaming happens somewhere in the Rails middleware.

Ok, fixed (that's a bit confusing).